### PR TITLE
Added packaging to cmake

### PR DESCRIPTION
**Pull request by cybercritter**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
+
+# Create a deployable installer. AKA: Packaging. We will use Red Hat Package Manager with CPack
+# Whether to build on fedora
+install(TARGETS build_system_playground LIBRARY DESTINATION lib64)
+SET(VERSION "${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}")
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_GENERATOR "RPM")
+set(CPACK_PACKAGE_NAME "build_system_playground")
+set(CPACK_PACKAGE_CONTACT "Michael J Reid")
+set(CPACK_PACKAGE_VENDOR "L3 Technologies")
```
Comment (amarcum) [inline CMakeLists.txt]: Should we set this to something else for github?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
 project (build_system_playground)
 
 
-#set(CMAKE_CXX_STANDARD 17)
+#compiler options setup
+#set(CMAKE_CXX_STANDARD 14)
 option(MULTITHREAD_SUPPORT_ENABLED "Multithreaded Support Enabled" TRUE)
-SET( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -std=c++1z" )
-add_library(build_system_playground SHARED src/build_system_playground/main.cpp)
+set( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -std=c++1z" )
```
Comment (amarcum) [inline CMakeLists.txt:16]: add_definitions("-std=c++1z") is another way this can be added, Just FYI.

Comment (cybercritter) [reply CMakeLists.txt:16]: Yeah. but this is more explicit as to adding it to the flags. CMake 3.8.0 adds c++ 17 support to the CMAKE_CXX_STANDARD macro. It is not released yet though. :(

File: CMakeLists.txt (hunk 1 above)
Comment (cybercritter) [reply CMakeLists.txt]: yes. I goofed it. I have pushed an update

File: CMakeLists.txt (hunk 2 above)
Comment (amarcum) [reply CMakeLists.txt:16]: Sounds good.

**Pull request merged**